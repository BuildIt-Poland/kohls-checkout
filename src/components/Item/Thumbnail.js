import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { spacingSmall, zIndexDefault } from '../../styles/designTokens';

const Wrapper = styled.div`
  position: relative;
  width: 100%;
  padding: ${spacingSmall};
  height: 14.4rem;
`;

const Image = styled.img`
  width: 100%;
`;

const Overlay = styled.div`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0, 160, 160, 0.1);
  z-index: ${zIndexDefault};
`;

function Thumbnail({ src, alt }) {
  return (
    <Wrapper>
      <Overlay />
      <Image src={src} alt={alt} />
    </Wrapper>
  );
}

Thumbnail.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired
};

export default Thumbnail;
