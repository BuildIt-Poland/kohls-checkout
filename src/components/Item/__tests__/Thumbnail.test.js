import React from 'react';
import { create } from 'react-test-renderer';

import Thumbnail from '../Thumbnail';

describe('COMPONENT - Item Thumbnail', () => {
  it('renders correctly', () => {
    const component = create(<Thumbnail src="http://taylor.swift.com/pic.jpg" alt="Very nice image" />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
