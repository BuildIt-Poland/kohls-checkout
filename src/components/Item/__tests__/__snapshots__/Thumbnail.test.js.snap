// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Item Thumbnail renders correctly 1`] = `
.c0 {
  position: relative;
  width: 100%;
  padding: 1rem;
  height: 14.4rem;
}

.c2 {
  width: 100%;
}

.c1 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0,160,160,0.1);
  z-index: 1;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <img
    alt="Very nice image"
    className="c2"
    src="http://taylor.swift.com/pic.jpg"
  />
</div>
`;
